slsDetectorSoftware/src/DetectorImpl.cpp

```diff
@@ -28,177 +28,173 @@
 
 namespace sls {
 
-DetectorImpl::DetectorImpl(int multi_id, bool verify, bool update)
-    : multiId(multi_id), multi_shm(multi_id, -1) {
-    setupMultiDetector(verify, update);
+DetectorImpl::DetectorImpl(int detector_index, bool verify, bool update)
+    : detectorIndex(detector_index), shm(detector_index, -1) {
+    setupDetector(verify, update);
 }
 
 DetectorImpl::~DetectorImpl() = default;
 
-void DetectorImpl::setupMultiDetector(bool verify, bool update) {
+void DetectorImpl::setupDetector(bool verify, bool update) {
     initSharedMemory(verify);
     initializeMembers(verify);
     if (update) {
         updateUserdetails();
     }
 }
 
-void DetectorImpl::setAcquiringFlag(bool flag) {
-    multi_shm()->acquiringFlag = flag;
-}
+void DetectorImpl::setAcquiringFlag(bool flag) { shm()->acquiringFlag = flag; }
 
-int DetectorImpl::getMultiId() const { return multiId; }
+int DetectorImpl::getDetectorIndex() const { return detectorIndex; }
 
-void DetectorImpl::freeSharedMemory(int multiId, int detPos) {
+void DetectorImpl::freeSharedMemory(int detectorIndex, int detPos) {
     // single
     if (detPos >= 0) {
-        SharedMemory<sharedSlsDetector> temp_shm(multiId, detPos);
-        if (temp_shm.IsExisting()) {
-            temp_shm.RemoveSharedMemory();
+        SharedMemory<sharedModule> moduleShm(detectorIndex, detPos);
+        if (moduleShm.IsExisting()) {
+            moduleShm.RemoveSharedMemory();
         }
         return;
     }
 
-    // multi - get number of detectors from shm
-    SharedMemory<sharedMultiSlsDetector> multiShm(multiId, -1);
-    int numDetectors = 0;
+    // multi - get number of modules from shm
+    SharedMemory<sharedDetector> detectorShm(detectorIndex, -1);
+    int numModules = 0;
 
-    if (multiShm.IsExisting()) {
-        multiShm.OpenSharedMemory();
-        numDetectors = multiShm()->numberOfDetectors;
-        multiShm.RemoveSharedMemory();
+    if (detectorShm.IsExisting()) {
+        detectorShm.OpenSharedMemory();
+        numModules = detectorShm()->numberOfModules;
+        detectorShm.RemoveSharedMemory();
     }
 
-    for (int i = 0; i < numDetectors; ++i) {
-        SharedMemory<sharedSlsDetector> shm(multiId, i);
-        shm.RemoveSharedMemory();
+    for (int i = 0; i < numModules; ++i) {
+        SharedMemory<sharedModule> moduleShm(detectorIndex, i);
+        moduleShm.RemoveSharedMemory();
     }
 }
 
 void DetectorImpl::freeSharedMemory() {
     zmqSocket.clear();
-    for (auto &d : detectors) {
+    for (auto &d : modules) {
```

Comment:
Maybe auto &m : modules? or module : modules?

Reply:
changed all the d's to module